### PR TITLE
609 bug fix baseline snmpcheck

### PR DESCRIPTION
**Pull request by CAfitech**
Bugs/Issues:

Interfaces did not have appropriate docstring.
The FormValuesType interface was located inside the main component.
Every function did not have appropriate docstring.
Every React hook did not have an appropriate inline comment.
Every component constants did not have appropriate inline comments.
The title consisted of more than the tools name.
The description was overly lengthy and wasn't just from the source documentation.
The steps were included in the description instead of separately.
The steps were not included or formatted.
The sourceLink was not included.
The tutorial was not included in the code.
The dependencies were not included.
The form did not have clear defined comments.
useEffect was not incorporated in the code.
All of the dependencies components were not installed.
Handle Process Data was not incorporated in the code.
Handle Process Termination was not incorporated in the code.
The onSubmit function did not have clear defined comments and docstring
The clearOutput was not clearly defined.
The outputV2 was not defined neither incorporated in the code.
The Render component was not incorporated.
Fixes:

Added all Interfaces with appropriate docstring.
Moved the FormValuesType interface so its located outside the main component.
Modified all the functions to have appropriate docstring.
Modified so that all React hook have appropriate in line comments.
Modified so that all component constants have appropriate in line comments.
Modified the title to only include the name of the tool.
Modified the description to be concise and accurate as per the checklist.
Modified the code by adding the steps which are those only applicable to the feature.
Modified the steps to properly format them.
Modified to add the sourceLink containing the link to the source component.
Modified to add the tutorial with an empty string.
Modified to add the dependencies with the required packages for the tool.
Modified so that the form is clearly defined including comments.
Modified to incorporate the useEffect, clearly defining including comments.
Modified to install the dependencies required.
Modified to incorporate Handle Process Data with clearly defined docstring and comments
Modified to incorporate Handle Process Termination with clearly defined docstring and comments
Modified the onSubmit function to include clearly defined docstrings and comments
Modified to incorporate the outputV2 clearly defined and in use.
Modified to include the Render Component within the return of the component function.

### CODE REVIEW

**Comment by CAfitech** [comment]
Ready for review

File: src/components/SnmpCheck/SnmpCheck.tsx [hunk 1]
```diff
@@ -1,75 +1,115 @@
...
+ * @field port: The SNMP port, defaults to 161 if not specified.
+ */
+interface FormValuesType {
     ip: string;
     port: number;
 }
+// Component Constants.
+const title = "SnmpCheck"; // Title of the tool.
+const description = // Description of the tool.
+    "The SNMP Check tool enables you to perform SNMP (Simple Network Management Protocol) checks on a specific IP " +
```
Comment (michaeljpigott) [inline src/components/SnmpCheck/SnmpCheck.tsx:24]: Be consistent with the naming of the tool. Is it SnmpCheck or SNMP Check?

File: src/components/SnmpCheck/SnmpCheck.tsx [hunk 2]
```diff
@@ -1,75 +1,115 @@
...
+    "address or hostname and port.";
+const steps =
+    "Step 1: Enter the IP address or hostname of the target device\n" +
```
Comment (michaeljpigott) [inline src/components/SnmpCheck/SnmpCheck.tsx:27]: Need a full stop at the end of this sentence.

File: src/components/SnmpCheck/SnmpCheck.tsx [hunk 3]
```diff
@@ -1,75 +1,115 @@
...
+    "Step 2: (Optional) Specify a target port number (default port: 161).\n" +
+    "Step 3: Click the 'Scan' button to initiate the SNMP check.\n";
+const sourceLink = "https://www.kali.org/tools/snmpcheck/"; // Link to the source code.
+const tutorial = ""; // Link to the official documentation/tutorial.
+const dependencies = ["snmpcheck"]; // Contains the dependencies required by the component.
```
Comment (michaeljpigott) [inline src/components/SnmpCheck/SnmpCheck.tsx:32]: Pretty sure this should be snmp-check.

**Review by michaeljpigott — CHANGES_REQUESTED** [review]
Hi CAfitech

This is looking good. Just a couple of grammatical suggestions and please check if you are using the right command in the dependencies constant. 

Kind regards 

Michael

File: src/components/SnmpCheck/SnmpCheck.tsx (hunk 1 above)
Comment (CAfitech) [reply src/components/SnmpCheck/SnmpCheck.tsx:24]: Hi Modified to make it consistent.

File: src/components/SnmpCheck/SnmpCheck.tsx (hunk 2 above)
Comment (CAfitech) [reply src/components/SnmpCheck/SnmpCheck.tsx:27]: I have included the full stop at the end of the sentence.

File: src/components/SnmpCheck/SnmpCheck.tsx (hunk 3 above)
Comment (CAfitech) [reply src/components/SnmpCheck/SnmpCheck.tsx:32]: yeah I have changed it.

**Review by michaeljpigott — APPROVED** [review]

**Pull request merged**